litex/tools/litex_soc_gen.py

```diff
@@ -62,56 +63,66 @@ def build(self, fragment, build_dir, build_name, **kwargs):
 # LiteX SoC Generator ------------------------------------------------------------------------------
 
 class LiteXSoCGenerator(SoCMini):
-    def __init__(self, name="litex_soc", sys_clk_freq=int(50e6), **kwargs):
+    def __init__(self, name="litex_soc", sys_clk_freq=int(50e6), n_mi=3, n_si=0, **kwargs):
+        assert (n_mi + n_si) < 5
+        self.kwargs = kwargs
```

Comment:
Not sure this is required, it seems to be used to get the bus standard, but you can get it with `self.bus.standard`.

Reply:
Okay, I will revise the code and submit another version to you.